tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -54,64 +53,68 @@ internal static CallTargetState StartInvocationOneParameter<TArg>(ILambdaExtensi
         internal static CallTargetReturn<TReturn> EndInvocationSync<TReturn>(TReturn returnValue, Exception exception, Scope scope, ILambdaExtensionRequest requestBuilder)
         {
             var json = SerializeObject(returnValue);
-            scope?.Dispose();
             Flush();
-            NotifyExtensionEnd(requestBuilder, exception != null, json);
+            NotifyExtensionEnd(requestBuilder, scope, exception != null, json);
+            scope?.Dispose();
             return new CallTargetReturn<TReturn>(returnValue);
         }
 
         internal static TReturn EndInvocationAsync<TReturn>(TReturn returnValue, Exception exception, Scope scope, ILambdaExtensionRequest requestBuilder)
         {
             var json = SerializeObject(returnValue);
-            scope?.Dispose();
             Flush();
-            NotifyExtensionEnd(requestBuilder, exception != null, json);
+            NotifyExtensionEnd(requestBuilder, scope, exception != null, json);
+            scope?.Dispose();
             return returnValue;
         }
 
         internal static CallTargetReturn EndInvocationVoid(Exception exception, Scope scope, ILambdaExtensionRequest requestBuilder)
         {
             scope?.Dispose();
             Flush();
-            NotifyExtensionEnd(requestBuilder, exception != null);
+            NotifyExtensionEnd(requestBuilder, scope, exception != null);
             return CallTargetReturn.GetDefault();
         }
 
-        internal static Scope CreatePlaceholderScope(Tracer tracer, ILambdaExtensionRequest requestBuilder)
+        internal static Scope CreatePlaceholderScope(Tracer tracer, string traceId, string samplingPriority)
         {
-            Scope scope = null;
-            try
+            Span span = null;
+            if (traceId == null || samplingPriority == null)
             {
```

Comment:
If `traceId != null` but `samplingPriority == null`, then we will enter this branch. Shouldn't we use the provided `traceId` in that case for calculating the sampling priority and for creating the new `Span`?

Reply:
✅ good catch thanks